index.js

```diff
@@ -112,56 +124,58 @@ function exec(str, cols, opts) {
 			if (rowLength) {
 				rows.push('');
 			}
-			wrapWord(rows, word, cols);
+			wrapWord(rows, x, cols);
 			continue;
 		}
 
 		if (rowLength + lengths[i] > cols && rowLength > 0) {
 			if (options.wordWrap === false && rowLength < cols) {
-				wrapWord(rows, word, cols);
+				wrapWord(rows, x, cols);
 				continue;
 			}
 
 			rows.push('');
 		}
 
 		if (rowLength + lengths[i] > cols && options.wordWrap === false) {
-			wrapWord(rows, word, cols);
+			wrapWord(rows, x, cols);
 			continue;
 		}
 
-		rows[rows.length - 1] += word;
+		rows[rows.length - 1] += x;
 	}
 
 	pre = rows.map(x => x.trim()).join('\n');
 
-	for (let j = 0; j < pre.length; j++) {
-		const y = pre[j];
+	for (const item of Array.from(pre).entries()) {
+		const i = item[0];
+		const x = item[1];
 
-		ret += y;
+		ret += x;
 
-		if (ESCAPES.indexOf(y) !== -1) {
-			const code = parseFloat(/\d[^m]*/.exec(pre.slice(j, j + 4)));
+		if (ESCAPES.indexOf(x) !== -1) {
+			const code = parseFloat(/\d[^m]*/.exec(pre.slice(i, i + 4)));
 			escapeCode = code === END_CODE ? null : code;
 		}
 
-		const code = ESCAPE_CODES.get(parseInt(escapeCode, 10));
+		const code = ESCAPE_CODES.get(Number(escapeCode));
 
 		if (escapeCode && code) {
-			if (pre[j + 1] === '\n') {
+			if (pre[i + 1] === '\n') {
 				ret += wrapAnsi(code);
-			} else if (y === '\n') {
+			} else if (x === '\n') {
 				ret += wrapAnsi(escapeCode);
 			}
 		}
 	}
 
 	return ret;
-}
+};
 
 // For each newline, invoke the method separately
 module.exports = (str, cols, opts) => {
 	return String(str)
+		.normalize()
```

Comment:
To normalize unicode characters that creates the same characters but with different code points and possibly different lengths.